### PR TITLE
Feature/ti2023 14/login functionality(backend)

### PR DESCRIPTION
**Pull request by bObXm**

### CODE REVIEW

File: EventPlannerAPI/BusinessLayer/Profiles/UserProfileForLogIn.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BusinessLayer.Profiles
+{
+    public class UserProfileForLogIn:Profile
+    {
+ /*       public UserProfileForLogIn()
```
Comment (alexandratodea) [inline EventPlannerAPI/BusinessLayer/Profiles/UserProfileForLogIn.cs]: remove commented code if not used

File: EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs [hunk 2]
```diff
@@ -8,5 +8,8 @@ public interface IUserService
 		Task<IdentityResult> CreateUserAsyncLogic(UserDto user);
         Task<string> SendPasswordResetLinkAsync(ForgotPasswordDto forgotPasswordDto);
         Task<IdentityResult> SetNewPasswordAsync(SetNewPasswordDto setNewPasswordDto);
+        string GenerateTokenString(LogInUserDto logInUserDto);
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs]: Why is only string?

File: EventPlannerAPI/BusinessLayer/Profiles/UserProfileForLogIn.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using AutoMapper;
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Profiles/UserProfileForLogIn.cs]: Remove class if is not used

File: EventPlannerAPI/BusinessLayer/Services/UserService.cs [hunk 4]
```diff
@@ -87,5 +91,40 @@ public async Task<IdentityResult> SetNewPasswordAsync(SetNewPasswordDto setNewPa
...
+                return await _userRepository.LogInAsync(userDto.UserIdentifier, userDto.Password);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, "An error occurred while logging in");
+                throw;
+            }
+        }
+
+        public string GenerateTokenString(LogInUserDto logInUserDto)
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs]: Move this function in a service for Authentification

File: EventPlannerAPI/DataAccessLayer/Contexts/EventPlannerContext.cs [hunk 5]
```diff
@@ -14,6 +14,8 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
             base.OnModelCreating(modelBuilder);
         }
+
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Contexts/EventPlannerContext.cs]: Remove empty space

File: EventPlannerAPI/DataAccessLayer/Contexts/EventPlannerContext.cs [hunk 6]
```diff
@@ -14,6 +14,8 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Contexts/EventPlannerContext.cs]: Same

File: EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs [hunk 7]
```diff
@@ -16,16 +16,36 @@ public UserController(IUserService userService)
...
+        {
+            if (!ModelState.IsValid)
+            {
+                var errors = ModelState.Values.SelectMany(v => v.Errors).Select(e => e.ErrorMessage);
+                return BadRequest(errors);
+            }
+            var resposnse = await _userService.LogInAsync(logInUserDto);
+            if (!resposnse)
+            {
+                return BadRequest();
```
Comment (HellHound97) [inline EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs]: Why no message in bad request?

File: EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs [hunk 8]
```diff
@@ -4,5 +4,7 @@ namespace DataAccessLayer.Models
 {
     public class EventPlannerUser : IdentityUser
     {
+
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs]: Same

File: EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs [hunk 9]
```diff
@@ -4,5 +4,7 @@ namespace DataAccessLayer.Models
...
+
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs]: Same

File: EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs [hunk 10]
```diff
@@ -4,5 +4,7 @@ namespace DataAccessLayer.Models
 {
     public class EventPlannerUser : IdentityUser
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: Why is this class empty?

File: EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs [hunk 11]
```diff
@@ -52,6 +53,20 @@ public async Task<IdentityResult> CreateUserAsync (EventPlannerUser user, string
             }
         }
 
+        public async Task<bool> LogInAsync(string userIdentifier, string password)
+        {
+               
+            var userByEmail = await _userManager.FindByEmailAsync(userIdentifier);
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs]: Why do you have 2 search methods?
Create one that search by UserName and Email.

File: EventPlannerAPI/EventPlannerAPI/Program.cs [hunk 12]
```diff
@@ -47,6 +50,26 @@
...
+    options.TokenValidationParameters = new TokenValidationParameters()
+    {
+        ValidateActor = true,
+        ValidateIssuer = true,
+        ValidateAudience = true,
+        RequireExpirationTime = true,
+        ValidateIssuerSigningKey = true,
+        ValidIssuer = builder.Configuration.GetSection("Jwt:Issuer").Value,
+        ValidAudience = builder.Configuration.GetSection("Jwt:Audience").Value,
+        IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration.GetSection("Jwt:Key").Value))
```
Comment (HellHound97) [inline EventPlannerAPI/EventPlannerAPI/Program.cs:68]: Why i don't see JWT:KEY value in this PR?

File: EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs (hunk 10 above)
Comment (bObXm) [reply EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: This class will help in the fututre when we want to make relationships between the user and other objects (events, etc)
For example:
public class EventPlannerUser: IdentityUser
{
    public virtual ICollection<Event> Events { get; set; }
}


and of course make setup up the relationship in the onModelCreating function inside the dbContext

somthing like:

 builder.Entity <Event> ()
            .HasOne(h => h.EventAppUser)
            .WithMany(u => u.Events)
            .HasForeignKey(h => h.UserId)
            .IsRequired();

File: EventPlannerAPI/EventPlannerAPI/Program.cs (hunk 12 above)
Comment (bObXm) [reply EventPlannerAPI/EventPlannerAPI/Program.cs:68]: Jwt:Key is not included for security reasons and it is stored in a secure location: appsettings.Local.json

Comment (HellHound97) [reply EventPlannerAPI/EventPlannerAPI/Program.cs:68]: And the other persons don't need it?

File: EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs (hunk 10 above)
Comment (HellHound97) [reply EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: It is used now?

Comment (bObXm) [reply EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: Know I realize the we need it

Comment (HellHound97) [reply EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: Remove it

Comment (bObXm) [reply EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: We need this class
Doesn't work without the class

Comment (HellHound97) [reply EventPlannerAPI/DataAccessLayer/Models/EventPlannerUser.cs:5]: okey

**Review by alexandratodea — APPROVED** [review]

**Pull request merged**